avm-transpiler/src/utils.rs

```diff
@@ -15,18 +15,34 @@ pub fn extract_brillig_from_acir_program(program: &Program) -> &[BrilligOpcode]
     assert_eq!(
         program.functions.len(),
         1,
-        "An AVM program should have only a single ACIR function flagged as 'BrilligCall'"
-    );
-    let opcodes = &program.functions[0].opcodes;
-    assert_eq!(
-        opcodes.len(),
-        1,
-        "An AVM program should have only a single ACIR function flagged as 'BrilligCall'"
+        "An AVM program should have only a single ACIR function with a 'BrilligCall'"
     );
+    let main_function = &program.functions[0];
+    let opcodes = &main_function.opcodes;
     match opcodes[0] {
-        Opcode::BrilligCall { .. } => {}
+        Opcode::BrilligCall { id, .. } => assert_eq!(id, 0, "The ID of the `BrilligCall` must be 0 as we have a single `Brillig` function"),
         _ => panic!("Tried to extract a Brillig program from its ACIR wrapper opcode, but the opcode doesn't contain Brillig!"),
     }
+    // Noir currently defaults to making a witness for each return witness in order
+    // to match the return types serialization. The current way the compiler does this is by
+    // generating a constraint for each return witness, which is what is checked below.
+    // We should not expect any other opcodes for transpilation.
+    if opcodes.len() > 1 {
```

Comment:
@dbanks12 @Maddiaa0 After this PR (https://github.com/noir-lang/noir/pull/4951) we have moved to making all ACIR functions use `distinct` on default. We currently do this in a somewhat inefficient way that generates a new constraint for each new witness. This may be improved in the future, but currently the check for a single BrilligCall opcode needed to be updated as it does not represent the program exactly. This should not affect actually transpiling the Brillig bytecode to AVM bytecode, but rather just these extraction checks.